R/p2_indices.R

```diff
@@ -0,0 +1,455 @@
+#' @title Calculate Climatological Monthly and Annual Indices
+#' 
+#' @description
+#' This function calculates various climatological monthly and annual climate indices
+#' using daily climate data.The indices include mean temperature, temperature anomalies, precipitation totals,
+#' precipitation anomalies, standardized precipitation index (SPI), and various temperature percentiles.
+#' 
+#' @param df A data.frame containing daily climate data.
+#' @param station The name of the station column in the data frame.
+#' @param date The name of the date column in the data frame.
+#' @param precip The name of the precipitation/rain column in the data frame.
+#' @param tmax The name of the maximum temperature column in the data frame.
+#' @param tmin The name of the minimum temperature column in the data frame.
+#' @param qct Quality Control level for temperature data (0, 1, or 2).
+#' @param qcpr Quality Control level for precipitation data (0, 1, or 2).
+#' @param nybr The start year for the climatological period.
+#' @param nyer The end year for the climatological period.
+#' @param station_df A data.frame containing station metadata.
+#' @param name The name of the station column in station data frame (to link with df[[station]]).
+#' @param lat The name of the latitude column in station data frame.
+#' @param lon The name of the longitude column in station data frame.
+#'
+#' @return A data frame containing calculated monthly and annual climate indices.
+#' 
+#' @export
+#'
+#' @examples
+#' # Example usage:
+#' # indices_result <- p2_indices(df = climate_data, station = "StationID", date = "Date",
+#'                             precip = "Precipitation", tmax = "MaxTemperature",
+#'                             tmin = "MinTemperature", qcpr = 2, nybr = 1981, nyer = 2010,
+#'                             station_df = station_metadata, name = "StationName",
+#'                             lat = "Latitude", lon = "Longitude")
+#' 
+#' @details
+#' This function is adapted from the ET-NCMP/NCMP project (https://github.com/ET-NCMP/NCMP) and modified for use in R-Instat.
```

Comment:
Looks great - just the same two comments as the other PR.

1. Can you put this as a link, and put `@details` to be before `@examples`. 

To put as a link - 

\href{https://github.com/ET-NCMP/NCMP}{ET-NCMP/NCMP}